File: Utilities/DatabaseUpdater/Global.vb
Public Class [Global]
	Private Shared _globalLock As New Object
	Private Shared _connectionString As String
	Shared Property ConnectionString() As String
		Get
			Return _connectionString
		End Get
		Set(ByVal value As String)
			SyncLock (_globalLock)
				_connectionString = value
			End SyncLock
		End Set
	End Property
End Class

File: Utilities/SqlScriptRunner/Global.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace SqlScriptRunner
{
    internal  class Global
    {
        private static object _globalLock = new object();
        private static string _connectionString;

        internal static string ConnectionString
        {
            get
            {
                return _connectionString;
            }
            set
            {
                lock ((_globalLock))
                {
                    _connectionString = value;
                }
            }
        }

		internal static System.Data.SqlClient.SqlConnection GetConnection(){
			return new System.Data.SqlClient.SqlConnection (ConnectionString);
		}

        internal static DateTime ToolStarted { get; set; }
    }
}
